src/containers/BuyManaWithFiatModal/BuyManaWithFiatFeedbackModal/BuyManaWithFiatFeedbackModal.tsx

```diff
@@ -1,34 +1,110 @@
+import { getAnalytics } from '../../../modules/analytics/utils'
+import { Purchase, PurchaseStatus } from '../../../modules/mana/types'
 import { t } from '../../../modules/translation/utils'
 import { Props } from './BuyManaWithFiatFeedbackModal.types'
 
+const transactionStatuses = {
+  [PurchaseStatus.PENDING]: TransactionStatus.PENDING,
+  [PurchaseStatus.COMPLETE]: TransactionStatus.SUCCESS,
+  [PurchaseStatus.FAILED]: TransactionStatus.FAILURE,
+  [PurchaseStatus.CANCELLED]: TransactionStatus.FAILURE
+}
+
+const camelToSnakeCase = (str: string) =>
+  str.replace(/[A-Z]/g, letter => `_${letter.toLowerCase()}`)
+
+const propsToTranslateByStatus = {
+  [TransactionStatus.PENDING]: [
+    'title',
+    'statusTitle',
+    'description',
+    'goToText'
+  ],
+  [TransactionStatus.SUCCESS]: ['title', 'description', 'cta'],
+  [TransactionStatus.FAILURE]: [
+    'title',
+    'statusTitle',
+    'description',
+    'cta',
+    'secondaryCta'
+  ]
+}
+
 const getDefaultFeedbackTranslations = (
-  network: Network
-): FeedbackModalI18N => ({
-  title: t(
-    `@dapps.buyManaWithFiat.feedback_modal.${network.toLowerCase()}.title`
-  ),
-  description: t(
-    `@dapps.buyManaWithFiat.feedback_modal.${network.toLowerCase()}.description`
-  ),
-  cta: t(`@dapps.buyManaWithFiat.feedback_modal.${network.toLowerCase()}.cta`),
-  error: ''
-})
+  { network, gateway }: Purchase,
+  status: TransactionStatus
+): FeedbackModalI18N => {
+  const basePath = `@dapps.buyManaWithFiat.feedback_modal.${status}`
+  return Object.fromEntries(
+    propsToTranslateByStatus[status].map(it => [
+      it,
+      t(`${basePath}.${camelToSnakeCase(it)}`, {
+        network: networksNames[network],
+        gateway: gatewaysNames[gateway]
+      })
+    ])
+  ) as FeedbackModalI18N
+}
 
 const BuyManaWithFiatFeedbackModal = ({
-  metadata: { purchase },
+  metadata: { purchase, goToUrl },
+  onTryAgain,
+  onSelectOtherProvider,
   onClose
 }: Props) => {
+  const { network, gateway, status: purchaseStatus } = purchase
+  const transactionStatus = transactionStatuses[purchaseStatus]
+  const analytics = getAnalytics()
+
+  const handleCtaClick = () => {
+    switch (transactionStatus) {
+      case TransactionStatus.SUCCESS:
+        onClose()
+        break
+
+      case TransactionStatus.FAILURE:
+        analytics.track('Try again with same Gateway', { network, gateway })
+        onTryAgain(network, gateway)
+        onClose()
+        break
+
+      default:
+        break
+    }
+  }
+
+  const handleSecondaryCtaClick = () => {
+    if (transactionStatus === TransactionStatus.FAILURE) {
+      analytics.track('Select other gateway', {
+        network,
+        previousGateway: gateway
+      })
+      onSelectOtherProvider(network)
+      onClose()
+    }
+  }
+
   return (
     <BaseBuyManaWithFiatFeedbackModal
       open
+      status={transactionStatus}
+      selectedNetwork={purchase.network}
```

Comment:
Now it might be a little bit too late, but I think it could have been cleaner to send the whole `purchase` as prop and the modal could have read the fields from it (network, gateway, tx status, etc). Just a comment in case you wanna refactor it later (not needed for this PR)

Reply:
It was a conscious decision, maybe not the best. The UI doesn't understand purchases, so I thought about mapping the purchase status, for example, into the transaction status that the modal understands.